File: net/pdfjet/TextUtils.cs
/*
 * TextUtils.cs
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
using System;
using System.Globalization;

namespace PDFjet.NET {
/// <summary>The helper that the examples use to print how long each one took.</summary>
public class TextUtils {
    /// <summary>Prints the name of an example and how long it took.</summary>
    public static void PrintDuration(String example, long time0, long time1) {
        String duration = ((time1 - time0)/1.0).ToString("F1", CultureInfo.InvariantCulture);
        if (duration.Length == 3) {
            duration = "    " + duration;
        } else if (duration.Length == 4) {
            duration = "   " + duration;
        } else if (duration.Length == 5) {
            duration = "  " + duration;
        } else if (duration.Length == 6) {
            duration = " " + duration;
        }
        Console.WriteLine(example + " => " + duration);
    }
}   // End of TextUtils.cs
}   // End of namespace PDFjet.NET

File: src/textutils.go
// textutils.go
//
// Copyright (c) 2026 PDFjet Software
// Licensed under the MIT License. See LICENSE file in the project root.

package pdfjet

import (
	"fmt"
)

// PrintDuration prints how long an example took: the difference between the
// two times, in milliseconds.
func PrintDuration(example string, time0, time1 int64) {
	duration := fmt.Sprintf("%d", time1-time0)
	if len(duration) == 1 {
		duration = "    " + duration
	} else if len(duration) == 2 {
		duration = "   " + duration
	} else if len(duration) == 3 {
		duration = "  " + duration
	} else if len(duration) == 4 {
		duration = " " + duration
	}
	duration += ".0"
	fmt.Println(example + " => " + duration)
}
